Subject: i386: prepare sharing the PIT code

PIT clock events work already and the PIT handling is the same for
i386 and x86_64. x86_64 does not support PIT as a clock source, so
disable the PIT clocksource for x86_64.

Prepare i8253.h to be shared with x8664

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Ingo Molnar <mingo@elte.hu>

---
 arch/i386/kernel/i8253.c |    4 +++-
 include/asm-i386/i8253.h |    6 ++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

Index: linux-2.6.23-rt1/arch/i386/kernel/i8253.c
===================================================================
--- linux-2.6.23-rt1.orig/arch/i386/kernel/i8253.c	2007-10-11 15:58:15.000000000 -0400
+++ linux-2.6.23-rt1/arch/i386/kernel/i8253.c	2007-10-11 15:59:49.000000000 -0400
@@ -13,7 +13,6 @@
 #include <asm/delay.h>
 #include <asm/i8253.h>
 #include <asm/io.h>
-#include <asm/timer.h>
 
 DEFINE_SPINLOCK(i8253_lock);
 EXPORT_SYMBOL(i8253_lock);
@@ -120,6 +119,7 @@ void __init setup_pit_timer(void)
 	global_clock_event = &pit_clockevent;
 }
 
+#ifndef CONFIG_X86_64
 /*
  * Since the PIT overflows every tick, its not very useful
  * to just read by itself. So use jiffies to emulate a free
@@ -204,3 +204,5 @@ static int __init init_pit_clocksource(v
 	return clocksource_register(&clocksource_pit);
 }
 arch_initcall(init_pit_clocksource);
+
+#endif
Index: linux-2.6.23-rt1/include/asm-i386/i8253.h
===================================================================
--- linux-2.6.23-rt1.orig/include/asm-i386/i8253.h	2007-10-11 15:58:15.000000000 -0400
+++ linux-2.6.23-rt1/include/asm-i386/i8253.h	2007-10-11 15:59:49.000000000 -0400
@@ -1,8 +1,6 @@
 #ifndef __ASM_I8253_H__
 #define __ASM_I8253_H__
 
-#include <linux/clockchips.h>
-
 /* i8253A PIT registers */
 #define PIT_MODE		0x43
 #define PIT_CH0			0x40
@@ -10,8 +8,12 @@
 
 extern spinlock_t i8253_lock;
 
+#ifdef CONFIG_GENERIC_CLOCKEVENTS
+
 extern struct clock_event_device *global_clock_event;
 
 extern void setup_pit_timer(void);
 
+#endif
+
 #endif	/* __ASM_I8253_H__ */
